src/server/db_slice.cc

```diff
@@ -598,60 +602,58 @@ void DbSlice::FlushSlots(SlotSet slot_ids) {
   }).Detach();
 }
 
-void DbSlice::FlushDb(DbIndex db_ind) {
+void DbSlice::FlushDbIndexes(const std::vector<DbIndex>& indexes) {
   // TODO: to add preeemptiveness by yielding inside clear.
-
-  if (db_ind != kDbAll) {
-    auto& db = db_arr_[db_ind];
+  DbTableArray flush_db_arr(db_arr_.size());
+  for (DbIndex index : indexes) {
+    auto& db = db_arr_[index];
     if (db) {
-      InvalidateDbWatches(db_ind);
-    }
+      InvalidateDbWatches(index);
+      flush_db_arr[index] = std::move(db);
 
-    auto db_ptr = std::move(db);
-    DCHECK(!db);
-    CreateDb(db_ind);
-    db_arr_[db_ind]->trans_locks.swap(db_ptr->trans_locks);
+      CreateDb(index);
+      db_arr_[index]->trans_locks.swap(flush_db_arr[index]->trans_locks);
+      if (TieredStorage* tiered = shard_owner()->tiered_storage(); tiered) {
+        tiered->CancelAllIos(index);
+      }
+    }
+  }
 
-    auto cb = [this, db_ptr = std::move(db_ptr)]() mutable {
-      if (db_ptr->stats.tiered_entries > 0) {
-        for (auto it = db_ptr->prime.begin(); it != db_ptr->prime.end(); ++it) {
-          if (it->second.IsExternal()) {
-            PerformDeletion(it, shard_owner(), db_ptr.get());
+  auto cb = [this, flush_db_arr = std::move(flush_db_arr)]() mutable {
+    for (auto& db_ptr : flush_db_arr) {
+      if (db_ptr) {
+        if (db_ptr->stats.tiered_entries > 0) {
```

Comment:
may I suggest to make if (db_ptr && db_ptr->stats.tiered_entries > 0) and instead of 2 scopes we will have one